wave_lang/kernel/wave/utils/graph_utils.py

```diff
@@ -212,6 +212,46 @@ def propagate_placeholders(n: fx.Node) -> fx.Node:
     return n
 
 
+def propagate_loop_carried_vars(n: fx.Node, depth: int = 0) -> fx.Node:
+    """
+    Propagates node through placeholders and loop-carried vars.
+
+    `depth` is the number of iterations to propagate through.
+    For example, if `depth` is 1, then the function will propagate through the
+    loop-carried vars to the previous iteration.
+    If `depth` is 0, then the function will propagate to the loop init_args.
+    """
```

Comment:
I don't understand what this function is doing. Do you just want to propagate to the init args from the iter args? (If so can you use get_inputs) Or go from get result to outputs?